src/plugins/kibana_react/public/page_template/solution_nav/solution_nav.tsx

```diff
@@ -35,69 +58,108 @@ export const KibanaPageTemplateSolutionNav: FunctionComponent<KibanaPageTemplate
   name,
   icon,
   items,
+  isOpenOnDesktop = false,
+  onCollapse,
   ...rest
 }) => {
-  const [isSideNavOpenOnMobile, setisSideNavOpenOnMobile] = useState(false);
+  // The EuiShowFor and EuiHideFor components are not in sync with the euiBreakpoint() function :(
+  const isSmallerBreakpoint = useIsWithinBreakpoints(['xs', 's']);
+  const isMediumBreakpoint = useIsWithinBreakpoints(['m']);
+  const isLargerBreakpoint = useIsWithinBreakpoints(['l', 'xl']);
+
+  // This is used for both the EuiSideNav and EuiFlyout toggling
+  const [isSideNavOpenOnMobile, setIsSideNavOpenOnMobile] = useState(false);
   const toggleOpenOnMobile = () => {
-    setisSideNavOpenOnMobile(!isSideNavOpenOnMobile);
+    setIsSideNavOpenOnMobile(!isSideNavOpenOnMobile);
   };
 
+  const isHidden = isLargerBreakpoint && !isOpenOnDesktop;
+
   /**
-   * Create the avatar.
+   * Create the avatar
    */
   let solutionAvatar;
   if (icon) {
     solutionAvatar = <KibanaPageTemplateSolutionNavAvatar iconType={icon} name={name} />;
   }
 
   /**
-   * Create the required title.
-   * a11y: Since the heading can't be nested inside `<nav>`, we have to hook them up via `aria-labelledby`
+   * Create the titles
    */
-  const titleID = htmlIdGenerator('kbnPageTemplateSolutionNav__title')();
-  const solutionNavTitle = (
-    <EuiTitle size="xs" id={titleID} className="kbnPageTemplateSolutionNav__title">
-      <h2>
-        {solutionAvatar}
-        <strong>{name}</strong>
-      </h2>
-    </EuiTitle>
+  const titleText = (
+    <>
+      {solutionAvatar}
+      <strong>{name}</strong>
+    </>
+  );
+  const mobileTitleText = (
+    <FormattedMessage
+      id="kibana-react.solutionNav.mobileTitleText"
+      defaultMessage="{solutionName} Menu"
+      values={{ solutionName: name || 'Navigation' }}
+    />
   );
 
   /**
    * Create the side nav component
    */
   let sideNav;
   if (items) {
-    const mobileTitleText = (
-      <FormattedMessage
-        id="kibana-react.pageTemplate.solutionNav.mobileTitleText"
-        defaultMessage="{solutionName} Menu"
-        values={{ solutionName: name || 'Navigation' }}
-      />
-    );
+    const sideNavClasses = classNames('kbnPageTemplateSolutionNav', {
+      'kbnPageTemplateSolutionNav--hidden': isHidden,
+    });
 
     sideNav = (
       <EuiSideNav
-        aria-labelledby={titleID}
+        aria-hidden={isHidden}
+        className={sideNavClasses}
+        heading={titleText}
         mobileTitle={
-          <h2>
+          <>
             {solutionAvatar}
             {mobileTitleText}
-          </h2>
+          </>
         }
         toggleOpenOnMobile={toggleOpenOnMobile}
         isOpenOnMobile={isSideNavOpenOnMobile}
-        items={items}
+        items={isHidden ? negativeTabIndex(items) : items}
         {...rest}
       />
     );
   }
 
   return (
-    <div className="kbnPageTemplateSolutionNav">
-      {solutionNavTitle}
-      {sideNav}
-    </div>
+    <>
```

Comment:
Some folks have suggested we rather use the full `React.Fragment` rather than `<>`. Does Eui dictate a preferred option?

Reply:
I tend to like Fragment myself because it's more explicit. I'll make that change.

Reply:
EUI doesn't have an official stance (as far as I know) but why are folks wanting to use `Fragment` instead of the shorthand?

Reply:
Shorthand gets lost when skimming code. There's no other reason.